packages/compass-components/src/components/drawer-portal.tsx

```diff
@@ -271,21 +273,77 @@ export const DrawerAnchor: React.FunctionComponent = ({ children }) => {
         return orderB < orderA ? 1 : orderB > orderA ? -1 : 0;
       });
   }, [drawerSectionItems]);
+
+  const [assistantNodes, setAssistantNodes] = useState<
+    Record<string, HTMLButtonElement | undefined>
+  >({});
+
+  const [failedLookupCount, setFailedLookupCount] = useState(0);
+
+  useEffect(
+    function () {
+      const nodes: Record<string, HTMLButtonElement | undefined> = {};
+      for (const [index, item] of toolbarData.entries()) {
+        if (!item.guideCue) {
+          continue;
+        }
+
+        const button = document.querySelector<HTMLButtonElement>(
+          `[data-testid="lg-drawer-toolbar-icon_button-${index}"]`
+        );
+        if (button) {
+          nodes[item.id] = button;
+        } else {
+          // we don't re-render enough times for unit tests to pass and this
+          // forces it to keep re-trying until the node is found
+          if (failedLookupCount < 10) {
```

Comment:
There must be a better way to do this? It seems to only affect the tests - in Compass proper the node's there immediately.

Reply:
This obviously has the extremely edge case bug where if something somehow replaces that dom node it won't pick it up again 😆 I can go further down this hacky rabbit hole to reset the counter again..